planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp

```diff
@@ -145,41 +145,66 @@ void StartPlannerModule::updateData()
 
 bool StartPlannerModule::isExecutionRequested() const
 {
-  const Pose & current_pose = planner_data_->self_odometry->pose.pose;
-  const lanelet::ConstLanelets current_lanes = utils::getCurrentLanes(planner_data_);
-  const double lateral_distance_to_center_lane =
-    lanelet::utils::getArcCoordinates(current_lanes, current_pose).distance;
-
-  if (std::abs(lateral_distance_to_center_lane) < parameters_->th_distance_to_middle_of_the_road) {
-    return false;
+  if (isModuleRunning()) {
+    return true;
   }
 
-  const Pose start_pose = planner_data_->route_handler->getOriginalStartPose();
+  // Return false and do not request execution if any of the following conditions are true:
+  // - The start pose is on the middle of the road.
+  // - The vehicle has already arrived at the start position planner.
+  // - The vehicle has reached the goal position.
+  // - The vehicle is still moving.
   if (
-    tier4_autoware_utils::calcDistance2d(start_pose.position, current_pose.position) >
-    parameters_->th_arrived_distance) {
+    isStartPoseOnMiddleOfTheRoad() || isCloseToOriginalStartPose() || hasArrivedAtGoal() ||
+    isMoving()) {
     return false;
   }
 
-  // Check if ego arrives at goal
-  const Pose goal_pose = planner_data_->route_handler->getGoalPose();
-  if (
-    tier4_autoware_utils::calcDistance2d(goal_pose.position, current_pose.position) <
-    parameters_->th_arrived_distance) {
+  // Check if the goal is behind the ego vehicle within the same route segment.
+  if (IsGoalBehindOfEgoInSameRouteSegment()) {
+    RCLCPP_WARN_THROTTLE(
+      getLogger(), *clock_, 5000, "Start plan for a backward goal is not supported now");
     return false;
   }
 
-  if (current_state_ == ModuleStatus::RUNNING) {
-    return true;
-  }
+  return true;
+}
 
-  const bool is_stopped = utils::l2Norm(planner_data_->self_odometry->twist.twist.linear) <
-                          parameters_->th_stopped_velocity;
-  if (!is_stopped) {
-    return false;
-  }
+bool StartPlannerModule::isModuleRunning() const
+{
+  return current_state_ == ModuleStatus::RUNNING;
+}
 
-  return true;
+bool StartPlannerModule::isStartPoseOnMiddleOfTheRoad() const
```

Comment:
StartPose -> CurrentPose is better?

because "start_pose" in the code is not always current pose like when backward driving is need

Reply:
I appreciate such a comment to understand better. Thanks!

Reply:
changed in https://github.com/autowarefoundation/autoware.universe/pull/5504/commits/9771e0a10117ce2ce5904b0e11a644dbf2cd05df